lib/App/Monitoring/Plugin/CheckRaid/Plugins/arcconf.pm

```diff
@@ -193,7 +193,7 @@ sub parse_ctrl_config {
 		if (/^\s+---+/) {
 			if (my($s) = <$fh> =~ /^\s+(\S.+?)\s*?$/) {
 				$subsection = $s;
-				unless (<$fh> =~ /^\s+---+/) {
+				unless (<$fh> =~ /^\s*$/ or /^\s+---+/) {
```

Comment:
I'm not entirely sure if this should be made differently here.

Reply:
i'm not certain does the `or` work, so I would use regexp "or":

```perl
				unless (<$fh> =~ /^\s*$|^\s+---+/) {
```

or just combine:
```perl
				unless (<$fh> =~ /^(?:\s*|+---+)$/) {
```

Reply:
@glensc thanks for looking into the merge request. So changing

```
--- a/lib/App/Monitoring/Plugin/CheckRaid/Plugins/arcconf.pm
+++ b/lib/App/Monitoring/Plugin/CheckRaid/Plugins/arcconf.pm
@@ -193,7 +193,7 @@ sub parse_ctrl_config {
                if (/^\s+---+/) {
                        if (my($s) = <$fh> =~ /^\s+(\S.+?)\s*?$/) {
                                $subsection = $s;
-                               unless (<$fh> =~ /^\s*$/ or /^\s+---+/) {
+                               unless (<$fh> =~ /^(?:\s*|\s+---+)$/) {
                                        $this->parse_error($_);
                                }
                                next;
```

will actually make the testsuite not pass anymore.

```
ok 124 - plugin created: arcconf31000
arcconf: parse error:    -------------------------------------------------------- at t//../lib/App/Monitoring/Plugin/CheckRaid/Plugins/arcconf.pm line 35, <$fh> line 147.
ok 125 - check ran
ok 126 - status code set
not ok 127 - status code matches

#   Failed test 'status code matches'
#   at t/check_arcconf.t line 198.
#          got: '3'
#     expected: '0'
ok 128 - status message
arcconf: parse error:    -------------------------------------------------------- at t//../lib/App/Monitoring/Plugin/CheckRaid/Plugins/arcconf.pm line 35, <$fh> line 147.
ok 129 - controller structure
# Looks like you failed 1 test of 129.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/129 subtests 

Test Summary Report
-------------------
t/check_arcconf.t (Wstat: 256 Tests: 129 Failed: 1)
  Failed test:  127
  Non-zero exit status: 1
```

Reply:
Here is the part which should be parsed:
```
   --------------------------------------------------------$
   Connector information$
                     $
   --------------------------------------------------------$
```
(The `$` are added to simplify/visualize how the strings looks like outputted from `arcconf`. The output is as well stored accordingly for the testcase (See https://github.com/glensc/nagios-plugin-check_raid/pull/210/commits/eb42ecd69a5ecffe5aee3d003728b273e999cb08).

Reply:
With some debugging added here what seems to happen in the case as per above:
```
Section: Controller information
Subjection: Power Settings
Subjection: Cache Properties
Subjection: Green Backup Information
Subjection: Physical Drive Write Cache Policy Information
Subjection: maxCache Properties
Subjection: RAID Properties
Subjection: Controller Version Information
Subjection: Temperature Sensors Information
Subjection: Out Of Band Interface Settings
Subjection: PBSI
Subjection: MCTP
Subjection: Connector information
Subjection: Connector #0
arcconf: parse error:    -------------------------------------------------------- at t//../lib/App/Monitoring/Plugin/CheckRaid/Plugins/arcconf.pm line 35, <$fh> line 147.
Section: Array Information
Subjection: Array Logical Device Information
Subjection: Array Physical Device Information
Subjection: Array Logical Device Information
Subjection: Array Physical Device Information
Section: Logical device information
Subjection: Array Physical Device Information
Subjection: Array Physical Device Information
Section: Physical Device information
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Subjection: Device Phy Information
Subjection: Device Error Counters
Section: maxCache information
```

So the parsing falls over the format and tries to parse the "Connector #0" then as well as subsection.

Reply:
@glensc, https://github.com/glensc/nagios-plugin-check_raid/pull/210/commits/c16aac2eba150c1a33c988a9cc69bb49148150f6 now adds the logic to check next line for beeing an empty line possibly with spaces, then check the next line for the subsection "closer".